pkg/storkctl/backuplocation.go

```diff
@@ -0,0 +1,221 @@
+package storkctl
+
+import (
+	"fmt"
+	"io"
+
+	storkv1 "github.com/libopenstorage/stork/pkg/apis/stork/v1alpha1"
+	"github.com/portworx/sched-ops/k8s"
+	"github.com/spf13/cobra"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/genericclioptions"
+	"k8s.io/kubernetes/pkg/printers"
+)
+
+const (
+	backupLocationSubcommand = "backuplocation"
+	hiddenString             = "<HIDDEN>"
+)
+
+var s3BackupLocationColumns = []string{"NAME", "PATH", "ACCESS-KEY-ID", "SECRET-ACCESS-KEY", "REGION", "ENDPOINT", "SSL-DISABLED"}
+var azureBackupLocationColumns = []string{"NAME", "PATH", "STORAGE-ACCOUNT-NAME", "STORAGE-ACCOUNT-KEY"}
+var googleBackupLocationColumns = []string{"NAME", "PATH", "PROJECT-ID"}
+
+func newGetBackupLocationCommand(cmdFactory Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
+	var showSecrets bool
+	getBackupLocationCommand := &cobra.Command{
+		Use:     backupLocationSubcommand,
+		Aliases: []string{"bl"},
+		Short:   "Get BackupLocations",
+		Run: func(c *cobra.Command, args []string) {
+			var backupLocations *storkv1.BackupLocationList
+			if len(args) > 0 {
+				if cmdFactory.AllNamespaces() {
+					util.CheckErr(fmt.Errorf("a resource cannot be retrieved by name across all namespaces"))
+					return
+				}
+
+				backupLocations = new(storkv1.BackupLocationList)
+				for _, name := range args {
+					bl, err := k8s.Instance().GetBackupLocation(name, cmdFactory.GetNamespace())
+					if err == nil {
+						backupLocations.Items = append(backupLocations.Items, *bl)
+					} else {
+						util.CheckErr(err)
+						return
+					}
+				}
+			} else {
+				namespaces, err := cmdFactory.GetAllNamespaces()
+				if err != nil {
+					util.CheckErr(err)
+					return
+				}
+
+				var tempBackupLocations storkv1.BackupLocationList
+				for _, ns := range namespaces {
+					backupLocations, err := k8s.Instance().ListBackupLocations(ns)
+					if err != nil {
+						util.CheckErr(err)
+						return
+					}
+					tempBackupLocations.Items = append(tempBackupLocations.Items, backupLocations.Items...)
+
+				}
+				backupLocations = &tempBackupLocations
+			}
+
+			if len(backupLocations.Items) == 0 {
+				handleEmptyList(ioStreams.Out)
+				return
+			}
+
+			s3BackupLocations := &storkv1.BackupLocationList{}
+			azureBackupLocations := &storkv1.BackupLocationList{}
+			googleBackupLocations := &storkv1.BackupLocationList{}
+			unknownBackupLocations := &storkv1.BackupLocationList{}
+			for _, bl := range backupLocations.Items {
+				switch bl.Location.Type {
+				case storkv1.BackupLocationS3:
+					if !showSecrets && bl.Location.S3Config != nil {
+						bl.Location.S3Config.SecretAccessKey = hiddenString
```

Comment:
do we also need to <hidde>  access key -id as well
Same for other cloud provider Storage-Account-Name and Project-ID

Reply:
Only trying to hide the secret information here.